www/js/survey/multilabel/multi-label-ui.js

```diff
@@ -180,6 +203,9 @@ angular.module('emission.survey.multilabel.buttons',
       isOther = true
       ConfirmHelper.checkOtherOption(inputType, checkOtherOptionOnTap, $scope);
     }
+    // special check for programs
+    // TODO: make this part of the dynamic config
+    expandInputsIfNecessary(inputType, $scope.selected[inputType].value);
```

Comment:
would it be more clean to include the `    if (inputType == "MODE") {
        if (inputValue == "e-bike") {` statements within this block of code? as creating a new function call that is required to be called after every trip object may lead to security vulnerabilites, and unnecessary use of the stack. I'm interested as to why you chose to do it this way (having a function that gets called everytime, but it only affects the desired output if the internal if statements are met).

Reply:
also, I am not sure what it means to "make it part of the dynamic config", if you could help me understand :)

Reply:
wrt

> "may lead to security vulnerabilites"

I am not aware of any security vulnerability engendered by making a call to another function in the same file. Can you clarify?

wrt

> "unnecessary use of the stack"

it is absolutely true that there is additional overhead caused by a function call instead of inline code. However, with modern compilers, the penalty is vanishingly small. And pulling into out into a separate function helps with modularity and maintainability.

Concretely, it makes it easier to call from multiple locations (as in https://github.com/e-mission/e-mission-phone/pull/874/commits/aba9dfb242fcb8d04f33030a9162d9c454472d1d) without copy-pasting.

And when we moved the code around (as we did in the subsequent PRs), it was much easier to move this function than to potentially pull out inline code with unexpected side effects.